### PR TITLE
JFB-2

### PR DESCRIPTION
**Pull request by geo-tat**

### CODE REVIEW

**Comment by geo-tat** [comment]
Have a question:
1) What does mean "all Ticket values" in part#3 of our home task?

**Comment by AtticSiol** [comment]
> Have a question:
> 
> 1. What does mean "all Ticket values" in part#3 of our home task?

You need to have setters only for 2 fields (“time” and “stadium sector”) and getters for all of fields.

File: src/service/TicketService.java [hunk 1]
```diff
@@ -9,4 +9,7 @@ public interface TicketService {
     Ticket getById(String id);
     List<Ticket> gatAll();
```
Comment (AtticSiol) [inline src/service/TicketService.java]: *gEtAll

File: src/model/User.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package model;
+
+abstract public class User extends ID {
```
Comment (AtticSiol) [inline src/model/User.java]: User is not an ID (OOP inheritance principle). Rename the class ID or move this code to some interface, for example "Identifiable"

File: src/service/TicketService.java (hunk 1 above)
Comment (geo-tat) [reply src/service/TicketService.java]: done

File: src/model/User.java (hunk 2 above)
Comment (geo-tat) [reply src/model/User.java]: done

File: src/model/Ticket.java [hunk 3]
```diff
@@ -1,87 +1,110 @@
...
+        if (ticketPrice.compareTo(BigDecimal.ZERO) < 0) {
+            throw new IllegalArgumentException("Invalid argument");
+        }
+
+        this.ticketId = ticketId;
+        this.concertHall = concertHall;
+        this.eventCode = eventCode;
         this.time = time;
         this.isPromo = isPromo;
         setStadiumSector(stadiumSector);
```
Comment (AtticSiol) [inline src/model/Ticket.java]: if there are no additional logic in  setStadiumSector method, than you can just assign the value:
this.stadiumSector = stadiumSector;

File: src/model/Ticket.java [hunk 4]
```diff
@@ -1,87 +1,110 @@
...
 import java.time.format.DateTimeFormatter;
 import java.util.Currency;
 import java.util.Locale;
+import java.util.Objects;
 
-public class Ticket {
-    private String id;
-    private String concertHall;
-    private String eventCode;
+public class Ticket implements Printable,Identifiable {
```
Comment (AtticSiol) [inline src/model/Ticket.java]: can't find setter for time field. please check, perhaps it was accidentally removed

File: src/model/Ticket.java (hunk 3 above)
Comment (geo-tat) [reply src/model/Ticket.java]: done

File: src/model/Ticket.java (hunk 4 above)
Comment (geo-tat) [reply src/model/Ticket.java]: image
it was here :)

File: src/model/Ticket.java [hunk 5]
```diff
@@ -1,87 +1,110 @@
...
-    private double maxAllowedBackpackWeightInKg;
+    private final double maxAllowedBackpackWeightInKg;
     private final LocalDateTime creationTime = LocalDateTime.now();
-    private BigDecimal ticketPrice;
+    private final BigDecimal ticketPrice;
 
     private static final String DATA_TIME_FORMAT = "yyyy.MM.dd, HH:mm";
 
     public Ticket() {
+        this.classId = IdGenerator.generateId();
```
Comment (AtticSiol) [inline src/model/Ticket.java]: this "IdGenerator.generateId()" code is duplicated in each your class. If you use Interface, you can move this code into the interface Identifiable. Think about how to do it properly

Comment (geo-tat) [reply src/model/Ticket.java]: done

**Review by AtticSiol — APPROVED** [review]

**Pull request merged**